File: 20180710/Reshape the Matrix.java
class Solution {
    public int[][] matrixReshape(int[][] nums, int r, int c) {
        int origin_r = nums.length;
        if (origin_r < 1){
            return nums;
        }
        int origin_c = nums[0].length;
        if (origin_c * origin_r != r * c){
            return nums;
        }
        List<Integer> expand = new ArrayList();
        for(int i = 0; i < origin_r; i++){
            for (int j = 0; j < origin_c; j ++){
                expand.add(nums[i][j]);
            }
        }
        int[][] res = new int[r][c];
        
        for(int i = 0; i < r; i++){
            for (int j = 0; j < c; j ++){
                res[i][j] = expand.get(j + c * i);
            }
        }
        return res;
    }
}

File: 20180710/Reshape the Matrix.py
class Solution(object):
    def matrixReshape(self, nums, r, c):
        """
        :type nums: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        origin_r = len(nums)
        if origin_r < 1:
            return nums
        origin_c = len(nums[0])
        if origin_c * origin_r != r * c:
            return nums
        expand = [item for row in nums for item in row]
        return [[expand[col + ( c * row)] for col in range(c)] for row in range(r)]